http-clients/aws-crt-client/src/main/java/software/amazon/awssdk/http/crt/ConnectionHealthChecksConfiguration.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.http.crt;
+
+import java.time.Duration;
+import software.amazon.awssdk.annotations.SdkPublicApi;
+import software.amazon.awssdk.utils.Validate;
+
+/**
+ * Configuration that defines health checks for for all connections established by
+ * the{@link ConnectionHealthChecksConfiguration}.
+ */
+@SdkPublicApi
+public final class ConnectionHealthChecksConfiguration {
+    private final long minThroughputInBytesPerSecond;
+    private final Duration allowableThroughputFailureInterval;
+
+    private ConnectionHealthChecksConfiguration(DefaultConnectionHealthChecksConfigurationBuilder builder) {
+        this.minThroughputInBytesPerSecond = Validate.paramNotNull(builder.minThroughputInBytesPerSecond,
+                                                                   "minThroughputInBytesPerSecond");
+        this.allowableThroughputFailureInterval = Validate.isPositive(builder.allowableThroughputFailureIntervalSeconds,
+                                                                      "allowableThroughputFailureIntervalSeconds");
+    }
+
+    /**
+     * @return the minimum amount of throughput, in bytes per second, for a connection to be considered healthy.
+     */
+    public long minThroughputInBytesPerSecond() {
+        return minThroughputInBytesPerSecond;
+    }
+
+    /**
+     * @return How long, in seconds, a connection is allowed to be unhealthy before getting shut down.
```

Comment:
nit: `Duration` doesn't have a single unit so I don't think it would make sense to say "in seconds" here. We might want to add a note on the setter that it only supports seconds precision though

Reply:
Yup, copy paste error